### PR TITLE
Simpler generate_params

### PR DESCRIPTION
**Pull request by ChristopherMayes**

### CODE REVIEW

File: pytao/interface_commands.py [hunk 1]
```diff
@@ -203,14 +215,10 @@ def bunch1(tao, *, ele_id, which="model", ix_bunch="1", coordinate="", verbose=F
         return __execute(tao, cmd, as_dict, method_name='bunch1', cmd_type='real_array')
 
 
-def building_wall_list(tao, ix_section="", *, verbose=False, as_dict=True):
+def building_wall_list(tao, *, ix_section, verbose=False, as_dict=True):
```
Comment (hhslepicka) [inline pytao/interface_commands.py:218]: This one I would expect to be:
```
def building_wall_list(tao, ix_section, *, verbose=False, as_dict=True):
```

File: pytao/interface_commands.py [hunk 2]
```diff
@@ -794,23 +850,10 @@ def data_set_design_value(tao, *, verbose=False, as_dict=True):
     return __execute(tao, cmd, as_dict, method_name='data_set_design_value', cmd_type='None')
 
 
-def datum_create(tao, *, datum_name, data_type, ele_ref_name="", ele_start_name="", ele_name="", merit_type="", meas="0", good_meas="F", ref="0", good_ref="F", weight="0", good_user="T", data_source="lat", eval_point="END", s_offset="0", ix_bunch="0", invalid_value="0", spin_axis_n0_1="", spin_axis_n0_2="", spin_axis_n0_3="", spin_axis_l_1="", spin_axis_l_2="", spin_axis_l_3="", verbose=False, as_dict=True):
+def datum_create(tao, datum_name, data_type, *, ele_ref_name, ele_start_name, ele_name, merit_type, meas="0", good_meas="F", ref="0", good_ref="F", weight="0", good_user="T", data_source="lat", eval_point="END", s_offset="0", ix_bunch="0", invalid_value="0", spin_axis_n0_1="''", spin_axis_n0_2="''", spin_axis_n0_3="''", spin_axis_l_1="''", spin_axis_l_2="''", spin_axis_l_3="''", verbose=False, as_dict=True):
```
Comment (hhslepicka) [inline pytao/interface_commands.py:853]: The parser is wrong here for the default value... `"''"`

File: pytao/interface_commands.py [hunk 3]
```diff
@@ -831,17 +874,33 @@ def datum_create(tao, *, datum_name, data_type, ele_ref_name="", ele_start_name=
...
-    spin_axis%n0(3) : optional
-    spin_axis%l(1) : optional
-    spin_axis%l(2) : optional
-    spin_axis%l(3) : optional
+    spin_axis%n0(1) : optional, default=''
+    spin_axis%n0(2) : optional, default=''
+    spin_axis%n0(3) : optional, default=''
+    spin_axis%l(1) : optional, default=''
+    spin_axis%l(2) : optional, default=''
+    spin_axis%l(3) : optional, default=''
```
Comment (hhslepicka) [inline pytao/interface_commands.py:883]: If default is empty just leave as `optional`.

File: pytao/interface_commands.py [hunk 4]
```diff
@@ -2178,11 +2342,14 @@ def lat_list(tao, *, elements, who, ix_uni="1", ix_branch="0", which="model", fl
     ix_uni : default=1
     ix_branch : default=0
     which : default=model
-    flags : optional
+    flags : optional, default=-array_out
```
Comment (hhslepicka) [inline pytao/interface_commands.py:2345]: Nice!

File: pytao/interface_commands.py (hunk 1 above)
Comment (ChristopherMayes) [reply pytao/interface_commands.py:218]: This is correct:

```
!%% building_wall_list -----------------------
! List of building wall sections or section points
!
! Notes
! -----
! Command syntax:
!   python building_wall_list {ix_section}
! If {ix_section} is not present then a list of building wall sections is given.
! If {ix_section} is present then a list of section points is given
! 
!
! Parameters
! ----------
! ix_section : optional
```

Comment (ChristopherMayes) [reply pytao/interface_commands.py:218]: Oh, I suppose it needs to be: `ix_section=''`

File: pytao/interface_commands.py (hunk 3 above)
Comment (ChristopherMayes) [reply pytao/interface_commands.py]: I changed things and corrected this.

File: pytao/interface_commands.py (hunk 2 above)
Comment (ChristopherMayes) [reply pytao/interface_commands.py]: Optionals now default `=''`

**Pull request merged**